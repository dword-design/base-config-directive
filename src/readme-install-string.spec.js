import tester from '@dword-design/tester'
import testerPluginTmpDir from '@dword-design/tester-plugin-tmp-dir'
import { outputFile } from 'fs-extra'
import stealthyRequire from 'stealthy-require-no-leak'

export default tester(
  {
    'extra scripts': async function () {
      await outputFile(
        'package.json',
        JSON.stringify({
          baseConfig: {
            cdnExtraScripts: [
              '<script src="https://unpkg.com/mermaid/dist/mermaid.min.js"></script>',
            ],
          },
          name: 'foo-bar',
        })
      )
      expect(
        stealthyRequire(require.cache, () => require('./readme-install-string'))
      ).toMatchSnapshot(this)
    },
    async valid() {
      await outputFile('package.json', JSON.stringify({ name: 'foo-bar' }))
      expect(
        stealthyRequire(require.cache, () => require('./readme-install-string'))
      ).toMatchSnapshot(this)
    },
  },
  [testerPluginTmpDir()]
)
