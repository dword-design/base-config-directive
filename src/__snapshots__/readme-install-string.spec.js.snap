// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readme-install-string extra scripts 1`] = `
"## Install via a package manager

\`\`\`bash
# npm
$ npm install foo-bar

# Yarn
$ yarn add foo-bar
\`\`\`

Add to local directives:

\`\`\`js
<script>
import FooBar from 'foo-bar'

export default {
  directives: {
    FooBar,
  },
}
</script>
\`\`\`

Or register as a global directive:

\`\`\`js
import Vue from 'vue'
import FooBar from 'foo-bar'

Vue.directive('FooBar', FooBar)
\`\`\`

Or register as a plugin:

\`\`\`js
import Vue from 'vue'
import FooBar from 'foo-bar'

Vue.use(FooBar)
\`\`\`

## Install via CDN

\`\`\`html
<script src=\\"https://unpkg.com/vue\\"></script>
<script src=\\"https://unpkg.com/mermaid/dist/mermaid.min.js\\"></script>
<script src=\\"https://unpkg.com/foo-bar\\"></script>
\`\`\`"
`;

exports[`readme-install-string valid 1`] = `
"## Install via a package manager

\`\`\`bash
# npm
$ npm install foo-bar

# Yarn
$ yarn add foo-bar
\`\`\`

Add to local directives:

\`\`\`js
<script>
import FooBar from 'foo-bar'

export default {
  directives: {
    FooBar,
  },
}
</script>
\`\`\`

Or register as a global directive:

\`\`\`js
import Vue from 'vue'
import FooBar from 'foo-bar'

Vue.directive('FooBar', FooBar)
\`\`\`

Or register as a plugin:

\`\`\`js
import Vue from 'vue'
import FooBar from 'foo-bar'

Vue.use(FooBar)
\`\`\`

## Install via CDN

\`\`\`html
<script src=\\"https://unpkg.com/vue\\"></script>
<script src=\\"https://unpkg.com/foo-bar\\"></script>
\`\`\`"
`;
